### ISSUE TITLE
Automatically support importing 3rd party svelte files

### ISSUE THREAD

**Issue by jakedeichert** [jakedeichert/svelvet#49]
Related to #47
And related snowpack discussion: https://github.com/pikapkg/snowpack/issues/213

**Two ideas so far:**

1. Svelvet could auto create a `snowpack.config.js` file in the user's project dir if it doesn't exist.
2. Work with snowpack to support a way for us to inject a rollup plugin without requiring a `snowpack.config.js` file.
Example: `snowpack --plugin rollup-plugin-svelte` ?

**Comment by dangodev** [comment]
> 1. Svelvet could auto create a `snowpack.config.js` file in the user's project dir if it doesn't exist.

 💯 I like this idea! We should probably add support for this regardless, as there are several cases where you need to maintain multiple config files. Using this, you probably could point to some secret config inside `node_modules` like so:

```js
exec(`${snowpackLocation} --config ${path.resolve(__dirname, 'snowpack.config.js')}`)
```

etc.

> 2. Work with snowpack to support a way for us to inject a rollup plugin without requiring a `snowpack.config.js file`. Example: `snowpack --plugin rollup-plugin-svelte` ?

This may work, but 2 things here give me concern:

1. Rollup doesn’t support `--plugin` via CLI, which means that Snowpack would have to invent/maintain this (and there might be a reason Rollup doesn’t do this that we’re not aware of).
1. the moment that either you needed to configure an option, or the plugin maintainer decided to change the default options, wouldn’t make this workable anymore

As things stand right now I think it could work, but all it would take is either one change to Svelvet or `rollup-plugin-svelte` for this to stop working. But again, #1 should work, would work for other Snowpack cases, and would always make sure you have the power to manage the config you need, makes me personally lean in that direction

**Comment by FredKSchott** [comment]
Oh you know what, I hadn't realized that Svelvet wrapped Snowpack. That `--config` proposal makes a lot more sense now! 😅

**Comment by jakedeichert** [comment]
> 💯 I really like this idea! This would keep control within Svelvet, and not only relieve the user of maintenance, but Svelvet can also ensure it’s working for them which should mean less support for you. With this, we could hide a config within `node_modules`

> Oh you know what, I hadn't realized that Svelvet wrapped Snowpack. That `--config` proposal makes a lot more sense now! 😅

Glad you both agree with that original idea! I didn't explain it well before 😅 

> So Svelvet users never run Snowpack? Just flagging that in case they are supposed to be able to, because using `--config` inside of Svelvet would mean that users running `snowpack` directly might break without it.

Exactly - svelvet users **never** have to run snowpack. It's part of the compilation pipeline since we need to compile src > dist first in order for `snowpack --include 'dist/**/*'` to work :)

So ideally, [this line](https://github.com/jakedeichert/svelvet/blob/a15254f01625f025edb08f3b63f4651ce9f3652d/src/index.ts#L166) in svelvet we would be able to add `--config ${path.resolve(__dirname, 'snowpack.config.js')` just like @dangodev suggested above. That way, the config is invisible to the user and we have full control. I would also allow svelvet users to override this config file if they want to, but snowpack doesn't need to worry about that.

**Comment by jakedeichert** [comment]
Snowpack just released `--config` support in v1.6.0 thanks to @dangodev 🎉 

We can use @dangodev's [solution posted here](https://github.com/pikapkg/snowpack/pull/232#issuecomment-593212965) to automatically load a snowpack config file with the rollup svelte plugin enabled so that our users don't have to manually create that.

We should allow the ability to override the default config that we provide. Maybe if svelvet sees a `snowpack.config.js` we just use that one instead of our own 👍

**Comment by dangodev** [comment]
> We should allow the ability to override the default config that we provide. Maybe if svelvet sees a snowpack.config.js we just use that one instead of our own 👍

I think that could definitely work, but I’d strongly recommend using Svelte’s default Snowpack config by default, and only allowing a user to use their own if they specify a `--snowpack-config` CLI flag (or whatever you want to call it). Reason is I think there would be a lot of cases where users may want to have a `snowpack.config.js` file that they don’t want affecting Svelte. IMO Svelte managing its own Snowpack config that’s out-of-the-box perfectly optimized for its own setup is the best user story.

But that said, I definitely see some utility in people wanting to add other Rollup plugins or some other transformation to extend Svelte, and the idea is cool! But again, just my opinion: it shouldn’t automatically try and search to find all the potential Snowpack configs a user may/may not want to use (which would be a challenge in itself).

Aside from that, if a user wants to use their own Snowpack config to power Svelte, now Snowpack has some pretty handy warnings. So a user will see clear errors like:

```
Try installing rollup-plugin-svelte and adding it to Snowpack (https://www.snowpack.dev/#custom-rollup-plugins)
```

We try to make it as friendly as we can 🙂

**Comment by jakedeichert** [comment]
Agreed! `--snowpack-config <path>` definitely works for me. Maybe if `<path>` is not supplied, we default to `./snowpack.config.js`. Do you think that's a fine trade-off?

**Comment by dangodev** [comment]
Sounds good to me!

**Comment by FredKSchott** [comment]
![excited](https://media.giphy.com/media/xTiN0CNHgoRf1Ha7CM/giphy.gif)
